Wikipedia/Code/WebViewController.m

```diff
@@ -773,13 +773,16 @@ - (void)scrollToFragment:(NSString *)fragment animated:(BOOL)animated completion
         [self.webView getScrollViewRectForHtmlElementWithId:fragment
                                                  completion:^(CGRect rect) {
                                                      if (!CGRectIsNull(rect)) {
-                                                         [self.webView.scrollView wmf_safeSetContentOffset:CGPointMake(self.webView.scrollView.contentOffset.x, rect.origin.y + [self.webView iOS12yOffsetHack] + self.delegate.navigationBar.hiddenHeight)
-                                                                                                  animated:animated
-                                                                                                completion:^(BOOL finished){
-                                                                                                    if (completion) {
-                                                                                                        completion();
-                                                                                                    }
-                                                                                                }];
+                                                         dispatchOnMainQueueAfterDelayInSeconds(0.1, ^{
```

Comment:
the repro steps in the GIF seem fixed without the delay. What's the reason for it?

Reply:
iirc it occurs intermittently. have seen it first time using the link, but also when tapping back then using the link again

Reply:
frequency varied by device too - saw much more often on iPad mini 4

Reply:
(actual device)

Reply:
just updated the mini to beta 2 - lemme re-test...

Reply:
here it is on 13.1 beta 2 on the mini 4 w/o the delay:

![delay mov](https://user-images.githubusercontent.com/3143487/64452086-74862480-d09a-11e9-917d-336a104e1508.gif)